core/src/main/scala-3/better/files/ResourceScalaCompat.scala

```diff
@@ -0,0 +1,115 @@
+package better.files
+
+import scala.quoted.{Quotes, Expr}
+import scala.quoted.*
+
+private[files] trait ResourceScalaCompat {
+
+  /** Look up class resource files.
+    *
+    * This Resource looks up resources relative to the JVM class file for `T`, using
+    * [[https://docs.oracle.com/javase/10/docs/api/java/lang/Class.html#getResource(java.lang.String) Class#getResource]]. For example, if
+    * `com.example.ExampleClass` is given for `T`, then resource files will be searched for in the `com/example` folder containing
+    * `ExampleClass.class`.
+    *
+    * If you want to look up resource files relative to the call site instead (that is, you want a class to look up one of its own
+    * resources), use the `my` method instead.
+    *
+    * @example
+    *   {{{Resource.at[YourClass].url("config.properties")}}}
+    * @tparam T
+    *   The class, trait, or object to look up from. Objects must be written with a `.type` suffix, such as `Resource.at[SomeObject.type]`.
+    * @return
+    *   A Resource for `T`.
+    * @see
+    *   [[https://docs.oracle.com/javase/10/docs/api/java/lang/Class.html#getResource(java.lang.String) Class#getResource]]
+    */
+  inline def at[T]: Resource = ${ Macros.atStaticImpl[T] }
+
+  /** Look up class resource files.
+    *
+    * This Resource looks up resources from the given Class, using
+    * [[https://docs.oracle.com/javase/10/docs/api/java/lang/Class.html#getResource(java.lang.String) Class#getResource]]. For example, if
+    * `classOf[com.example.ExampleClass]` is given for `clazz`, then resource files will be searched for in the `com/example` folder
+    * containing `ExampleClass.class`.
+    *
+    * If you want to look up resource files relative to the call site instead (that is, you want your class to look up one of its own
+    * resources), use the `my` method instead.
+    *
+    * @example
+    *   {{{Resource.at(Class.forName("your.AppClass")).url("config.properties")}}}
+    *
+    * In this example, a file named `config.properties` is expected to appear alongside the file `AppClass.class` in the package `your`.
+    * @param clazz
+    *   The class to look up from.
+    * @return
+    *   A Resource for `clazz`.
+    * @see
+    *   [[https://docs.oracle.com/javase/10/docs/api/java/lang/Class.html#getResource(java.lang.String) Class#getResource]]
+    */
+  def at(clazz: Class[_]): Resource = new Resource {
```

Comment:
I've not implemented `atDymanic(Class)` macro here since it seems not needed IIUC.